packages/@okta/vuepress-site/docs/guides/saml-inline-hook/main/index.md

```diff
@@ -0,0 +1,327 @@
+---
+title: SAML assertion inline hook
+excerpt: Learn how to easily implement a SAML assertion inline hook
+layout: Guides
+---
+
+This guide provides a working example of an Okta SAML assertion inline hook. It uses the website [Glitch.com](https://glitch.com) to act as an external service to receive and respond to SAML assertion inline hook calls.
+
+---
+
+**Learning outcomes**
+
+* Understand the Okta SAML assertion inline hook calls and responses.
+* Implement a simple working example of a SAML assertion inline hook with a Glitch.com external service.
+* Preview and test the SAML assertion inline hook.
+
+**What you need**
+
+* [Okta Developer Edition organization](https://developer.okta.com/signup/)
+* [Glitch.com](https://glitch.com) project or account
+* A sample application that uses SAML to implement authentication. This guide works with the application in the Sample code section below.
+
+**Sample code**
+
+* [Okta SAML assertion inline hook Example](https://glitch.com/edit/#!/okta-inlinehook-samlhook)
+* [Spring Boot, SAML, and Okta](https://github.com/oktadev/okta-spring-boot-saml-example)
+
+---
+
+## About SAML assertion inline hook implementation
+
+The SAML assertion inline hook can be used to customize the authentication workflow that occurs between an application and the Okta org, which functions as the Identity Provider (IdP).
+
+This guide provides an end-to-end scenario using a SAML-authenticated application and an Okta org, and includes example code for an external service to respond to calls from a SAML assertion inline hook triggered during the authentication workflow.
+
+### The scenario
+
+In the following scenario, the external service code parses a request from Okta, evaluates the user name against a simple patient data store, and, if the user is part of the patient store, responds to Okta with a command to add a patient ID claim to the SAML assertion. If the user name is not part of the data store, no action is taken.
+
+At a high-level, the following workflow occurs:
```

Comment:
I'd find it helpful to single-source that inline hooks diagram here :)

Reply:
Thanks. I revised the workflow steps to use the generic hook call diagram.